File: examples/pizzas_static_go/domain/order.go
package domain

import (
	"time"
	"github.com/google/uuid"
)

type Order struct {
	ID        string    `json:"id"`
	CustomerName string `json:"customer_name"`
	Items []OrderItem `json:"items"`
	Status string `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func NewOrder(customerName string, items []OrderItem, status string) *Order {
	a := &Order{
		ID:        uuid.New().String(),
		CustomerName: customerName,
		Items: items,
		Status: status,
		CreatedAt: time.Now(),
		UpdatedAt: time.Now(),
	}
	return a
}

func (a *Order) Validate() error {
	if a.CustomerName == "" {
		return &ValidationError{Field: "customer_name", Message: "customer_name can't be blank"}
	}
	return nil
}

File: examples/pizzas_domain/pizzas_domain/lib/pizzas_domain/order/order.rb
require 'hecks/mixins/model'

module PizzasDomain
  class Order
    autoload :OrderItem, "pizzas_domain/order/order_item"
    autoload :Lifecycle, "pizzas_domain/order/lifecycle"

    include Hecks::Model

    attribute :customer_name
    attribute :items, default: [], freeze: true
    attribute :status, default: "pending"

    # State predicates — see lifecycle.rb for full state machine
    def pending?; status == "pending"; end
    def cancelled?; status == "cancelled"; end

    private

    def validate!
      raise ValidationError.new("customer_name can't be blank", field: :customer_name, rule: :presence) if customer_name.nil? || (customer_name.respond_to?(:empty?) && customer_name.empty?)
    end
  end
end
